orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/v2/transform/V2TrimConditionalsTransform.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipelinetemplate.v1schema.graph.v2.transform;
+
+import com.netflix.spinnaker.orca.pipelinetemplate.v2schema.V2PipelineTemplateVisitor;
+import com.netflix.spinnaker.orca.pipelinetemplate.v2schema.model.V2PipelineTemplate;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.StageDefinition;
+
+import java.util.stream.Collectors;
+
+public class V2TrimConditionalsTransform implements V2PipelineTemplateVisitor {
+
+  @Override
+  public void visitPipelineTemplate(V2PipelineTemplate pipelineTemplate) {
+    trimConditionals(pipelineTemplate);
+  }
+
+  private void trimConditionals(V2PipelineTemplate pipelineTemplate) {
```

Comment:
Can we write up some user-facing docs on how all this works from a user perspective?

Reply:
Something like a "12 steps of pipeline template processing" reference doc? Or something different?

Reply:
That could be useful, the key understanding I'm looking for is all the stuff a user can configure against an existing template, and how that changes what's rendered.

Reply:
Makes sense.